src/main/scala/is/hail/io/bgen/BgenRecord.scala

```diff
@@ -206,12 +208,14 @@ class BgenRecordV12(compressed: Boolean, nSamples: Int, tolerance: Double) exten
             lastProb -= p
           }
           sampleProbs(j) = lastProb
+
           // GT
           val gt = Genotype.unboxedGTFromUIntLinear(sampleProbs)
           if (gt != -1)
-            rvb.addInt(gt)
+            rvb.addInt(Call(gt))
```

Comment:
I think this constructor should have an explicit name: `Call.fromUnphasedDiploidIndex` (oof, a mouthful, but at least accurate).